tests/test_forgejo/test_forgejo.py

```diff
@@ -0,0 +1,117 @@
+    event_misc_hashes,
+    event_misc_headers,
+    event_misc_string,
+    event_misc_summary,
+    event_pull_request_body,
+    event_pull_request_hashes,
+    event_pull_request_headers,
+    event_pull_request_string,
+    event_pull_request_summary,
+    event_push_body,
+    event_push_hashes,
+    event_push_headers,
+    event_push_string,
+    event_push_summary,
+)
+
+
+@pytest.mark.parametrize(
+    "headers, body, summary, string, hashes",
+    [
+        pytest.param(
+            event_fork_headers,
+            event_fork_body,
+            event_fork_summary,
+            event_fork_string,
+            event_fork_hashes,
+            id="Testing schema for Forgejo 'fork' event",
+        ),
+        pytest.param(
+            event_issue_comment_headers,
+            event_issue_comment_body,
+            event_issue_comment_summary,
+            event_issue_comment_string,
+            event_issue_comment_hashes,
+            id="Testing schema for Forgejo 'issue_comment' event",
+        ),
+        pytest.param(
+            event_issues_headers,
+            event_issues_body,
+            event_issues_summary,
+            event_issues_string,
+            event_issues_hashes,
+            id="Testing schema for Forgejo 'issues' event",
+        ),
+        pytest.param(
+            event_pull_request_headers,
+            event_pull_request_body,
+            event_pull_request_summary,
+            event_pull_request_string,
+            event_pull_request_hashes,
+            id="Testing schema for Forgejo 'pull_request' event",
+        ),
+        pytest.param(
+            event_push_headers,
+            event_push_body,
+            event_push_summary,
+            event_push_string,
+            event_push_hashes,
+            id="Testing schema for Forgejo 'push' event",
+        ),
+        pytest.param(
+            event_misc_headers,
+            event_misc_body,
+            event_misc_summary,
+            event_misc_string,
+            event_misc_hashes,
+            id="Testing schema for Forgejo 'misc' event",
+        ),
+    ],
+)
+def test_forgejo_events(headers, body, summary, string, hashes):
+    """
+    Test the Forgejo schema across various Forgejo events
+    """
+    mesg = ForgejoMessageV1(body={"body": body, "headers": headers, "agent": "testuser-w2fm"})
+
+    mesg.validate()
+    assert mesg.app_name == "Forgejo"
+    assert mesg.target_id == body["repository"]["id"]
+    assert mesg.target_type == "repository"
+    assert mesg.delivery == headers["x-forgejo-delivery"]
+    assert mesg.signature == hashes["sha160"]
+    assert mesg.signature_sha256 == hashes["sha256"]
+    assert mesg.event_name == headers["x-forgejo-event"]
+    assert mesg.summary == summary
+    assert (
+        mesg.app_icon == "https://apps.fedoraproject.org/img/icons/webhook-to-fedora-messaging.png"
+    )
+    assert mesg.usernames == ["testuser-w2fm"]
+    assert mesg.groups == []
+    for item in string:
```

Comment:
Why not `assert str(mesg) == string` ? Why use a list in "string" (which make it badly named, btw... ;-) )?

Reply:
Fair enough - I would change the variable name.

The reason why I am iterating through lines and not comparing them as-is is to avoid the pitfalls of newlines creeping in and causing annoying test failures.

Reply:
I don't see how newlines could creep in.